gen-json-schema-class.py

```diff
@@ -0,0 +1,225 @@
+        for slot in (cdef.get("slots") or []):
+            sdef = all_slots.get(slot)
+            if sdef:
+                used_slots[slot] = sdef
+                used_enums |= slot_enums(sdef)
+    collected_enums = {nm: all_enums[nm] for nm in used_enums if nm in all_enums}
+
+    header = yaml.safe_load(HEADER_YAML.read_text())
+    header.pop("classes", None)
+
+    yaml_content = {
+        **header,
+        "name": cls_name,
+        "classes": {
+            nm: {
+                "description": cdef.get("description"),
+                **({"is_a": cdef["is_a"]} if "is_a" in cdef else {}),
+                "slots": cdef.get("slots") or []
+            }
+            for nm, cdef in hierarchy
+        },
+        "slots": used_slots,
+        "enums": collected_enums
+    }
+
+    out_file = BUILD_DIR / f"{cls_name}.yaml"
+    out_file.write_text(yaml.safe_dump(yaml_content, sort_keys=False))
+    print(f"  • Wrote {out_file}")
+
+# ─── STEP 2 + 3: JSON SCHEMA GENERATION + DEREFERENCE ────────────────────────
+for tpl in BUILD_DIR.glob("*.yaml"):
+    cls = tpl.stem
+    print(f"\n🔨 Generating JSON Schema for {cls}")
+    schema_str = run_capture([
+        "gen-json-schema",
+        "--top-class", cls,
+        "--inline", "--no-metadata",
+        "--title-from=title",
+        "--closed", str(tpl)
+    ])
+    if not schema_str:
+        print(f"⚠️  Skipping {cls}")
+        continue
+
+    try:
+        raw = json.loads(schema_str)
+    except json.JSONDecodeError as e:
+        print(f"❌ Invalid JSON for {cls}: {e}")
+        continue
+
+    # set ID + title
+    raw["$id"]   = (
+        f"https://repo-prod.prod.sagebase.org/repo/v1/schema/type/registered/"
+        f"sagedm-{cls.lower()}"
+    )
+    raw["title"] = cls
+
+    # --- dereference, but leave $refs intact so we can inline enums ourselves
+    deref = jsonref.replace_refs(raw, merge_props=False, proxies=False)
+    full  = deref
+
+    # pull out all enum definitions
+    defs  = full.pop("$defs", {})
+    props = full.setdefault("properties", {})
+
+    # walk every property; when we hit a $ref to an enum, inline only its enum list + title
+    def resolve(obj):
+        if isinstance(obj, dict):
+            if "$ref" in obj:
+                ref       = obj.pop("$ref")
+                enum_name = ref.rsplit("/", 1)[-1]
+                enumdef   = defs.get(enum_name)
+                if enumdef and "enum" in enumdef:
+                    # merge only the enum values
+                    obj["enum"] = enumdef["enum"]
+                    # bring over human-friendly title if you set one
+                    if "title" in enumdef:
+                        obj["title"] = enumdef["title"]
+                return
+            for v in obj.values():
+                resolve(v)
+        elif isinstance(obj, list):
+            for item in obj:
+                resolve(item)
+
+    resolve(props)
+
+    # now normalize, order, and write out as before
```

Comment:
If use linkml==v1.8.1, don't need this per https://github.com/nf-osi/nf-metadata-dictionary/issues/622#issuecomment-3009674848